modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ConditionalProcessor.java

```diff
@@ -0,0 +1,442 @@
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.ConfigurationUtils;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+import org.elasticsearch.script.ProcessorConditionalScript;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptException;
+import org.elasticsearch.script.ScriptService;
+
+import static org.elasticsearch.ingest.ConfigurationUtils.newConfigurationException;
+import static org.elasticsearch.ingest.ConfigurationUtils.readMap;
+
+public class ConditionalProcessor extends AbstractProcessor {
+
+    static final String TYPE = "conditional";
+
+    private final Script condition;
+
+    private final ScriptService scriptService;
+
+    private final Processor processor;
+
+    private ConditionalProcessor(String tag, Script script, ScriptService scriptService, Processor processor) {
+        super(tag);
+        this.condition = script;
+        this.scriptService = scriptService;
+        this.processor = processor;
+    }
+
+    @Override
+    public void execute(IngestDocument ingestDocument) throws Exception {
+        if (scriptService.compile(condition, ProcessorConditionalScript.CONTEXT)
+            .newInstance(condition.getParams()).execute(new UnmodifiableIngestData(ingestDocument.getSourceAndMetadata()))) {
+            processor.execute(ingestDocument);
+        }
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        private final ScriptService scriptService;
+
+        public Factory(ScriptService scriptService) {
+            this.scriptService = scriptService;
+        }
+
+        @Override
+        public ConditionalProcessor create(Map<String, Processor.Factory> factories, String tag,
+            Map<String, Object> config) throws Exception {
+            Map<String, Map<String, Object>> processorConfig = readMap(TYPE, tag, config, "processor");
+            final Script script;
+            try (XContentBuilder builder = XContentBuilder.builder(JsonXContent.jsonXContent)
+                .map(normalizeScript(config.get("script")));
+                 InputStream stream = BytesReference.bytes(builder).streamInput();
+                 XContentParser parser = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
+                     LoggingDeprecationHandler.INSTANCE, stream)) {
+                script = Script.parse(parser);
+                config.remove("script");
```

Comment:
Can we have validation other keys do not exist?

Reply:
@rjernst I think we have that automatically upstream in the configuration utils, that's why I'm removing `script` here (we do that in the other processors too). If I don't remove it here the calling code throws

Reply:
Great! I didn't realize this was handled by the code that initializes the processors.